src/theory/arith/nl/nonlinear_extension.cpp

```diff
@@ -557,12 +557,15 @@ int NonlinearExtension::checkLastCall(const std::vector<Node>& assertions,
   }
   if (options::nlCad())
   {
-    lemmas = d_cadSlv.checkFull();
-    if (lemmas.empty())
+    d_cadSlv.checkFull();
+    if (!d_im.hasMadeProgress())
     {
       Trace("nl-cad") << "nl-cad found SAT!" << std::endl;
     }
-    filterLemmas(lemmas, wlems);
+    else
+    {
+      return 1;
```

Comment:
This doesnt fit with the return value of this method, I assume this will change later? This entire method is probably being heavily refactored and moved somewhere else I'm guessing? Maybe add a comment here.